### ISSUE TITLE
File-Pickers don't work in Chromium browsers.

### ISSUE THREAD

**Issue by unnamedcrewman** [hyprwm/Hyprland#436]
I'm using Hyprland on NixOS alongside Sway WM and GNOME as a Desktop Environment. 

As the issue suggests, when I'm in Hyprland, Chromium browsers are not giving me any kind of file-picker window for uploading local files in the browser.  All browsers give a file-picker in GNOME and Sway, though I've noticed that the file-picker Firefox uses is different than what the Chromium browsers use. The Firefox file-picker works as expected in Hyprland.

- **Steps to reproduce**:
    - Install a chromium browser (Brave, Google Chrome, ungoogled-chromium)
    - Attempt to upload a file from the local disk in one of those browsers
    - Eg. Add a file to a Bug Report on Github; Add an attachment in Gmail; Upload an assignment file in Moodle (learning mangagement platform) 
- **Expected Outcome**:
    - Clicking the Upload/Browse/Choose File link will open a file picker window.
- **Noted Outcome**:
    - No file-picker opens.

I'm not sure if `bug` is the best label here, but I wasn't sure how else to characterize it.

Thanks.

**Comment by fufexan** [comment]
Are you using the GNOME XDG Desktop Portal? It's known to interfere with the wlroots portal. I'd try putting Hyprland and GNOME into different specialisations just to test if it's the case.

**Comment by unnamedcrewman** [comment]
@fufexan Thansk for the quick reply.

I am using the XDG Desktop Portal at the moment, though that is simply because it's there by default with my setup.  I have to admit, I'm fairly new to Linux and I'm not sure how all this stuff interacts.  In particular, I'm not sure what you mean by "putting Hyprland and GNOME into different specializations".  Could you point me in the direction of some resources for learning about how to do that?

Thanks again.

**Comment by fufexan** [comment]
Specialisations are basically smaller subsystems in your entire system. They can contain completely opposite software which won't interact with each other, and that makes it ideal for quickly testing things like this one.

The option is described [here](https://search.nixos.org/options?channel=unstable&show=specialisation.%3Cname%3E.configuration&from=0&size=50&sort=relevance&type=packages&query=specialisations). You can read more about it [here](https://nixos.org/manual/nixos/stable/options.html#opt-specialisation).

**Comment by unnamedcrewman** [comment]
Thanks for the links, they look helpful and I'll get to reading them.  

I just disabled `xdg.portal` in my NixOS config and then uninstalled GNOME.  It worked, and the file-pickers are back for the chromium browsers, so I'm assuming that your assessment of the xdg.portal/wlroots conflict was correct.

I actually don't use GNOME and I've switched fully to Hyprland already, so I'm going to consider this closed, at least for my purposes.

Thanks again for your time.
